packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_use_cases/fetch_auth_token_use_case.dart

```diff
@@ -0,0 +1,20 @@
+import '../repositories/auth_repository.dart';
+
+class FetchAuthTokenUseCase {
+  FetchAuthTokenUseCase(
+    this._authRepository,
+  );
+
+  final AuthRepository _authRepository;
+
+  Future<String?> fetchNewAccessToken() async {
+    final refreshToken = await getRefreshToken();
+    if (refreshToken == null) {
+      return null;
+    }
+    final newToken = await _authRepository.fetchNewToken();
+    return newToken;
+  }
+
+  Future<String?> getRefreshToken() async => _authRepository.getRefreshToken();
```

Comment:
Is there the need of a separate method for fetching the refresh token? 
On line 11 we could just do `final refreshToken = await _authRepository.getRefreshToken();`

Reply:
The reason is that I take example from poolboy app. There all those methods, that depend of repo are extracted (abstracted), so the complex method doesn't know who delivers this information. But if you think it is better to move this logic in the main method (maybe because we have only one method like this), I will.